app/models/team_member.rb

```diff
@@ -0,0 +1,7 @@
+class TeamMember < ApplicationRecord
+  belongs_to :team
+  has_one :user
+
+  validates :role, presence: true
+  validates :team_id, presence: true
```

Comment:
The 'team_id' field is validated for presence, but there is no validation for its integrity, such as ensuring it references a valid 'team' record. Consider adding a validation to check that the 'team_id' corresponds to an existing team.